pilot/pkg/config/monitor/file_snapshotter.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitor
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+
+	"istio.io/istio/pilot/pkg/config/kube/crd"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/log"
+)
+
+var (
+	supportedExtensions = map[string]bool{
+		".yaml": true,
+		".yml":  true,
+	}
+)
+
+// FileSnapshotter holds a reference to a file directory that contains crd
+// config and filter criteria for which of those configs will be parsed.
+type FileSnapshotter struct {
+	root             string
+	configTypeFilter map[string]bool
+}
+
+// NewFileSnapshotter returns a snapshotter.
+// If no types are provided in the descriptor, all IstioConfigTypes will be allowed.
+func NewFileSnapshotter(root string, descriptor model.ConfigDescriptor) *FileSnapshotter {
```

Comment:
On naming: IMO 'snapshot' is an implementation detail, should not be part of the type name.
Maybe 'FileConfig' ? It may change in future to watching files, etc.

Reply:
This was a real hard object to name - it does take a snapshot of config at a point in time (aka when the monitor calls it). `FileConfig` feels really generic to me but I am open to renaming this as needed.